api/src/pcapi/core/banner/api.py

```diff
@@ -43,23 +43,34 @@ def _is_17_18_transition(user: users_models.User) -> bool:
     )
 
 
+def _has_completed_id_check(user: users_models.User) -> bool:
+    completed_id_check = next(
+        (
+            fraud_check
+            for fraud_check in user.beneficiaryFraudChecks
+            if fraud_check.type in (fraud_models.FraudCheckType.DMS, fraud_models.FraudCheckType.UBBLE)
+            and user.eligibility in fraud_check.applicable_eligibilities
+            and fraud_check.status == fraud_models.FraudCheckStatus.OK
+        ),
+        None,
+    )
+    return completed_id_check is not None
```

Comment:
(détail / style / question de goût) On utilise `next()` généralement quand on veut récupérer un élément d'un itérable. Ici ça n'est pas le cas, donc je pense qu'utiliser `any()` serait plus approprié, car plus clair : 

```suggestion
    return any(
        fraud_check
        for fraud_check in user.beneficiaryFraudChecks
        if fraud_check.type in (fraud_models.FraudCheckType.DMS, fraud_models.FraudCheckType.UBBLE)
        and user.eligibility in fraud_check.applicable_eligibilities
        and fraud_check.status == fraud_models.FraudCheckStatus.OK
    )
```